rpc/events.go

```diff
@@ -26,7 +26,7 @@ type EventsChunk struct {
 
 type EmittedEvent struct {
 	*Event
-	BlockNumber     uint64     `json:"block_number"`
-	BlockHash       *felt.Felt `json:"block_hash"`
+	BlockNumber     *uint64    `json:"block_number,omitempty"`
+	BlockHash       *felt.Felt `json:"block_hash,omitempty"`
```

Comment:
We should have a test case that checks that the response JSON doesn't have these values set, right?

Reply:
That would be testing the JSON marshaler, no? not our code

Reply:
Possibly, I just thought that we should have test case which shows that our pending event doesn't have `BlockNumber` and `BlockHash`

Reply:
I think there should be some checks on the single event returned in and of it self so that we can have something similar to `checkLatestBlock`. We should have at least one case for non-pending event and one for pending event (without block number and block hash).